### PR TITLE
[IMP] point_of_sale, pos_restaurant: improve split order name

### PR DESCRIPTION
**Pull request by AH-Yussef**
when an order is split from:
- a table, the new order name is (table number + char starting from b and increasing with more splits)
- a floating order, the new order name is (floating order name + char starting from b and increasing with more splits)

### CODE REVIEW

File: addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js:72]:
```suggestion
const sort_by_last_char = (a, b) => a.slice(-1).localeCompare(b.slice(-1));
```

File: addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js:59]:
```suggestion
        const filter_orders = (order) => order.getOrderName().slice(0, -1) === name
        };
```

File: addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js:84]:
```js
_getLatestOrderNameStartingWith(name) {
  return this.pos.get_open_orders()
    .map((order) => order.getOrderName())
    .filter((orderName) => orderName.slice(0, -1) === name)
    .sort((a, b) => a.slice(-1).localeCompare(b.slice(-1)))
    .at(-1) || name
}
```

File: addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js:86]: shouldn't we have upper case letters in this case ? `6A`, `6B`, instead of `6a`, `6b`

**Review by vlst-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_restaurant/static/src/overrides/models/pos_order.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/overrides/models/pos_order.js:22]: inline this code in `split_bill_screen.js`. It looks like it's only needed there.

File: addons/pos_restaurant/static/tests/tours/utils/chrome.js
Comment (vlst-odoo) [inline addons/pos_restaurant/static/tests/tours/utils/chrome.js:10]: typo

**Comment by caburj** [comment]
Hi @vlst-odoo @AH-Yussef. Can I trust you to merge this PR only after the following PR is merged?

https://github.com/odoo/odoo/pull/175535/files

**Pull request closed without merging**
